repos/system_upgrade/common/libraries/distro.py

```diff
@@ -16,3 +19,206 @@ def get_distribution_data(distribution):
+            'CentOS-Linux-BaseOS.repo': 'all',
+            'CentOS-Linux-ContinuousRelease.repo': 'all',
+            'CentOS-Linux-Debuginfo.repo': 'all',
+            'CentOS-Linux-Devel.repo': 'all',
+            'CentOS-Linux-Extras.repo': 'all',
+            'CentOS-Linux-FastTrack.repo': 'all',
+            'CentOS-Linux-HighAvailability.repo': 'all',
+            'CentOS-Linux-Media.repo': 'all',
+            'CentOS-Linux-Plus.repo': 'all',
+            'CentOS-Linux-PowerTools.repo': 'all',
+            'CentOS-Linux-Sources.repo': 'all',
+        },
+        '9': {
+            '/etc/yum.repos.d/centos.repo': 'all',
+            '/etc/yum.repos.d/centos-addons.repo': 'all',
+        },
+        '10': {
+            '/etc/yum.repos.d/centos.repo': 'all',
+            '/etc/yum.repos.d/centos-addons.repo': 'all',
+        },
+    },
+    'almalinux': {
+        '8': {
+            # TODO is this true on all archs?
+            '/etc/yum.repos.d/almalinux-ha.repo': 'all',
+            '/etc/yum.repos.d/almalinux-nfv.repo': 'all',
+            '/etc/yum.repos.d/almalinux-plus.repo': 'all',
+            '/etc/yum.repos.d/almalinux-powertools.repo': 'all',
+            '/etc/yum.repos.d/almalinux-resilientstorage.repo': 'all',
+            '/etc/yum.repos.d/almalinux-rt.repo': 'all',
+            '/etc/yum.repos.d/almalinux-sap.repo': 'all',
+            '/etc/yum.repos.d/almalinux-saphana.repo': 'all',
+            '/etc/yum.repos.d/almalinux.repo': 'all',
+        },
+        '9': {
+            '/etc/yum.repos.d/almalinux-appstream.repo': 'all',
+            '/etc/yum.repos.d/almalinux-baseos.repo': 'all',
+            '/etc/yum.repos.d/almalinux-crb.repo': 'all',
+            '/etc/yum.repos.d/almalinux-extras.repo': 'all',
+            '/etc/yum.repos.d/almalinux-highavailability.repo': 'all',
+            '/etc/yum.repos.d/almalinux-plus.repo': 'all',
+            '/etc/yum.repos.d/almalinux-resilientstorage.repo': 'all',
+            '/etc/yum.repos.d/almalinux-sap.repo': 'all',
+            '/etc/yum.repos.d/almalinux-saphana.repo': 'all',
+            # RT and NFV are only on x86_64 on almalinux 9
+            '/etc/yum.repos.d/almalinux-nfv.repo': ['x86_64'],
+            '/etc/yum.repos.d/almalinux-rt.repo': ['x86_64'],
+        },
+        '10': {
+            # no resilientstorage on 10
+            '/etc/yum.repos.d/almalinux-appstream.repo': 'all',
+            '/etc/yum.repos.d/almalinux-baseos.repo': 'all',
+            '/etc/yum.repos.d/almalinux-crb.repo': 'all',
+            '/etc/yum.repos.d/almalinux-extras.repo': 'all',
+            '/etc/yum.repos.d/almalinux-highavailability.repo': 'all',
+            '/etc/yum.repos.d/almalinux-plus.repo': 'all',
+            '/etc/yum.repos.d/almalinux-sap.repo': 'all',
+            '/etc/yum.repos.d/almalinux-saphana.repo': 'all',
+            # RT and NFV are only on x86_64 on almalinux 10
+            '/etc/yum.repos.d/almalinux-nfv.repo': ['x86_64'],
```

Comment:
for architectures, you please rather arch "constants" defined in the  `leapp.libraries.common.config.architecture` library. regarding all, what about using `ARCH_SUPPORTED`  instead? you can then simplify the logic. Also I would suggest to have one type of value, instead of str / list.

(note that `ARCH_SUPPORTED` is tuple)